src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -953,22 +953,16 @@ void MacroAssembler::pop_reg(Register Rd)
 }
 
 int MacroAssembler::bitset_to_regs(unsigned int bitset, unsigned char* regs) {
-  DEBUG_ONLY(int words_pushed = 0;)
-
   int count = 0;
-  // Sp is x2, and zr is x0, which should not be pushed.
-  // If the number of registers is odd, zr is used for stack alignment.Otherwise, it will be ignored.
-  bitset &= ~ (1U << 2);
-  bitset |= 0x1;
-
+  // Zr and sp (x0, x2) should not be pushed
+  assert((bitset & 0b101) == 0, "zr or sp is in bitset: %x", bitset);
```

Comment:
I don't think we need this assertion here. Function bitset_to_regs should not care about specific registers in the set.

Reply:
Removed as comment. Thanks.